Content.Server/Atmos/Piping/Unary/Components/GasVentPumpComponent.cs

```diff
@@ -59,6 +59,48 @@ public sealed partial class GasVentPumpComponent : Component
         [DataField("underPressureLockoutLeaking")]
         public float UnderPressureLockoutLeaking = 0.0001f;
 
+        #region some averaging stuff
```

Comment:
This averaging stuff could be broadly applicable to other things that need measurement. Consider making it its own helper class, though that is not a requirement.

Also, I know you have a 5 sample moving average, but how much real time does 5 samples correspond to? (Please document that).

Also consider if an exponential moving average (for an example, see the `GasPassiveGateSystem.cs`). It is a lot simpler to implement, doesn't need a cyclic buffer, and doesn't need to count the samples in the buffer.

Reply:
Forgot to update the name before pushing it, was thinking of using it to properly group the variables I was going to use. I don't think I'll turn it into its own helper class for now, I feel like it could be a bit too hard for me currently from a code design standpoint. 

Didn't know that exponential moving average is simpler to implement, kinda stopped thinking about it after hearing that Linux kernel uses it to report system usage for stuff like htop, and for now the kernel is black magic in my eyes.

Reply:
You can take a look at the code example I showed. EWMA is much less code.

Reply:
Assuming I'm reading the code properly, 5 samples should be like ~0.33 seconds, though it definitely doesn't feel like that. The tickrate should be specified in `atmos.tickrate`.

I'll proceed to test how this works with EWMA.